hierarchy.el

```diff
@@ -380,7 +384,7 @@ This function returns the result of applying FUNCTION to ITEM (the first
 root if nil)."
   (let ((item (or item (car (hierarchy-roots hierarchy))))
         (indent (or indent 0)))
-    (funcall function item indent
+    (funcall function item indent (map-elt (hierarchy--delaying-parents hierarchy) item)
```

Comment:
this modifies the contract for FUNCTION. I would prefer not doing that because it would impact all usages of the `hierarchy- map-tree`.

Reply:
Fair; it's been a while since I looked at this so I can't remember why I adjusted this. I remember trying to not jostle current functionality, as much as feasible. Lemme think on it and see if there's a way to pull that off.

Reply:
@DamienCassou, I'm not sure why I did this way, originally. Maybe I was figuring, just in case any casual user were to make use of `hierarchy-map-tree`, the designated parameter would force the user, who might be less familiar with the code, to remember that children functions would be returned to handle delayed children and not miss out the functionality because they didn't incorporate it into their `function` they pass.

I dunno; either way, no harm to the original logic I had and doesn't alter the number of parameters for the `function`. Updated here and the use in the tests.